### DISCUSSION TITLE
How to create dummy environment for Mirth TCP Sender?

### DISCUSSION THREAD

**Discussion by BforAzure2** [nextgenhealthcare/connect#4786] [Q&A] [answered]
Hello Mirth Geeks, 

We are generating a HL7 message and want to send to another EHR server in real time. We have configured a TCP Sender (MLLP) for that.   But before we place it on production, we want to test that our TCP sender is working fine. How can we create a dummy EHR environment where we receive this HL7 message from our own Mirth, get acknowledgement from dummy EHR listener and parse that message. Any solution or article for thi splease?

**Comment by pacmano1** [comment]
Just create a new channel that is a TCP listener.    

![image](https://user-images.githubusercontent.com/44065187/137175895-d7fae57f-9701-4bc0-9960-3f4bb59b1a01.png)

**Reply by BforAzure2** [reply]
@pacmano1 Thank you. But we want to place this listener on a new server and not the same Mirth engine. I was just reading about Soup HL7 and HAPI Test client programs. Can we use these for the said purpose?

Or, If we want to have a dummy EHR web portal, Is there any free access available where we can consume Mirth generated messages in an automated process?

**Reply by pacmano1** [reply]
You can use _anything_ that is capable of receiving your message running on some other server to which your Mirth engine can send traffic.   That's HL7 Soup, HL7 Spy, Mirth itself, etc.

All you will see in the tools mentioned is the HL7 message as you sent it.   I don't know of a "test" EHR that would consume your generated message and represet it in some pretty fashion other than your client's EHR.

**Reply by pacmano1** [reply]
I am not really clear on your objective.  If you have basic TCP connectivitiy between your server and some other box and have agreed on MLLP / HL7 formatting, it just works.

**Reply by BforAzure2** [reply]
Yes, I was wondering if we can create a similar EHR as of our client's EHR. 

Ok, I think we should go with HL7 Soup and receive our Mirth's message there to complete our test cycle. Thank you for your response. Highly appreciated.

**Reply by pacmano1** [reply]
Depending on the work you do, as part of your implementaion process you would be discussing your client's environment to determine if they have a test / developer instance.  It really depends on the space you are in.   For example for many practice managment systems, developers enroll in a developer program and get access to a test/dev instance for integration testing.  That instance is unrelated to your client and usually the EHR vendor will "certify" your soluion prior to allowing integration with their clients.

**Reply by BforAzure2** [reply]
@pacmano1 Let me give an overview of our space. We are generating a generic HL7 which will be consumed by many clients. The client's EHR can be Epic,Cerner or their In-house custom EHR.  Ideally, we want to test our proposed solution to all popular possible solutions, From EHR vendor to a simple desktop listener you mentioned above (HL7 Spy etc). 

So to put all this in perspective, we want to run test cycles at our end so that when we onboard our clients the experience should be seamless.

**Reply by pacmano1** [reply]
Well good luck with that depending on what transactions you are sending.     Not only do different EHRs want different formats, you may also find the same EHR at different clients will want different formats.    It is just a fact of the space so I would prepare for you work to be configuration table driven based on EHR type and client.   I am only touching the tip of the iceberg with the potential gotchas, there are even more of them if you are receiving inbound HL7 that in turn (parts of it) need to be returned back in your outbound messages.

**Reply by michaeljeffreycollins** [reply]
Even with a foundation driven delivery solution like you would see from EPIC 2 hospitals could have radically different requirements for a given HL7 message.  Main concerns would be field names and repeating fields.

**Comment by pacmano1** [comment, marked as answer]
Question answered in thread, OP did not flag one as answered.
